tasks.c

```diff
@@ -3866,6 +3868,7 @@ void vTaskSuspendAll( void )
     #else /* #if ( configNUMBER_OF_CORES == 1 ) */
     {
         UBaseType_t ulState;
+        const BaseType_t xCoreID = ( BaseType_t ) portGET_CORE_ID();
```

Comment:
The xCoreID must be acquired with interrupts disabled to prevent the task from being scheduled to another core.

Reply:
Is it sufficient to do this after the call to `portSET_INTERRUPT_MASK()`?

Reply:
Yes. The task can't be switched out with interrupt disabled.